File: data/code_pairs/fortran/2073.f90
module types
    implicit none

    type :: person
        character(len=20) :: forename
        integer :: age
    end type person

    type :: kids
        type(person) :: oldest_child
        type(person), allocatable, dimension(:) :: other_kids
    end type kids

    type, private :: AUXILIARY
        logical :: diagnostic
        character(len=20) :: message
    end type AUXILIARY

    type :: NUMERIC_SEQ
        integer :: int_val
        real :: real_val
        logical :: log_val
    end type NUMERIC_SEQ

    type :: POINT
        real :: x, y
    contains
        procedure :: length => point_length
    end type POINT

contains
    real function point_length(this)
        class(POINT), intent(in) :: this
        point_length = sqrt(this%x**2 + this%y**2)
    end function point_length
end module types

File: data/code_pairs/cpp/2073.cpp
#include <string>
#include <vector>
#include <cmath>

namespace types {

class person {
public:
    std::string forename;
    int age;
};

class kids {
public:
    person oldest_child;
    std::vector<person> other_kids;
};

class NUMERIC_SEQ {
public:
    int int_val;
    float real_val;
    bool log_val;
};

class POINT {
public:
    float x, y;

    float LENGTH() const {
        return std::sqrt(x*x + y*y);
    }
};

} // namespace types